# Does the Global South Receive Less Guardian Coverage as Crises Get More Serious?

## Table of Contents

1. Introduction

2. Data Acquisition

3. Preparation for Data Analysis

4. Exploratory Data Analysis

5. Conclusion

## 1. Introduction

The past decade has seen a succession of major conflict outbreaks — from the Syrian civil war and the Tigray conflict in Ethiopia, to the Russian invasion of Ukraine in 2022 and the escalation of violence in Sudan and Gaza in 2023. These crises have tested how global media allocates attention, and whether that allocation reflects the scale of human suffering or something else entirely.

Media coverage of global crises shapes public awareness and, in turn, political and humanitarian response. If a newspaper's coverage is disproportionately concentrated on Western conflicts, regardless of their relative severity, this has real implications for how its readership understands and prioritises global suffering. The Guardian is a particularly meaningful case study: it has an explicit editorial commitment to international reporting and positions itself as a progressive outlet with global reach. 

The concept of proximity bias in journalism is well-established in media studies literature: outlets tend to devote more coverage to geographically or culturally proximate events. However, existing work is largely qualitative or limited to specific conflicts. No publicly available analysis applies a severity-controlled, data-driven framework to Guardian coverage across a decade-long window. The originality of this project lies in using ACLED fatality data as a quantitative severity benchmark, allowing us to ask not just "who gets covered?" but "who gets covered relative to how serious their crisis is?"

### Research Questions

1. Is there a measurable disparity in Guardian coverage volume between Global South and Western countries, controlling for conflict severity?

2. Do structural features of articles, length and section placement, differ systematically by region?

3. Does The Guardian respond to casualty spikes differently depending on region, and has any disparity narrowed or widened over time (2015–2024)?


## 2. Data Acquisition

We use 3 main data sources:

- Guardian API (primary): Article metadata for all countries with recorded conflict, 2015–2024

- ACLED Data (supplementary): Reported fatalities by country-year, used as our conflict severity proxy, [Link here](https://acleddata.com/aggregated/number-reported-fatalities-country-year)

- Our World in Data / World Bank (supplementary): Country income classification, used to assign Global South / Western labels, [Link here](https://ourworldindata.org/grapher/world-bank-income-groups)

We first load in the appropriate libraries to start the data collection and cleaning. We also save each student's Guardian API key into a keys.json file.

In [1]:
import requests
import pandas as pd
import json
import time
import os
import numpy as np
from pathlib import Path
import plotly.graph_objects as go
from plotly.subplots import make_subplots
from scipy.stats import pearsonr

with open("../keys.json") as f:
    keys = json.load(f)

API_KEYS = [
    keys["guardian"]["api_key_1"],
    keys["guardian"]["api_key_2"],
    keys["guardian"]["api_key_3"]
]

JSONDecodeError: Expecting ':' delimiter: line 4 column 3 (char 27)

### 2.1 Guardian API and respective data collection

Once the correct librairies and API keys loaded, we query the Guardian's search endpoint for every country present in the ACLED dataset. Country names are converted to Guardian tag slugs programmatically, with a manual corrections dictionary for edge cases. USA, UK, and Australia are queried by section rather than tag, as they have dedicated top-level sections on the Guardian.

Hence, for each article we collect: publication date, section name, word count, article title, editorial pillar, and author byline.

Step 1: Loading in the ACLED file and converting the ACLED country names into Guardian API identifiable country tags.

In [ ]:
acled = pd.read_csv("../data/cleaned_fatalities.csv")

FROM_DATE = "2015-01-01"
TO_DATE   = "2024-12-31"

acled_countries = acled["COUNTRY"].unique().tolist()

def name_to_slug(name):
    return name.lower().strip().replace(" ", "-").replace("'", "").replace(",", "")

auto_slugs = {country: name_to_slug(country) for country in acled_countries}

SLUG_CORRECTIONS = {
    "Democratic Republic of Congo": "democratic-republic-of-congo",
    "Republic of Congo":            "republic-of-congo",
    "United States":                "us-news",
    "Myanmar":                      "myanmar",
    "Ivory Coast":                  "ivory-coast",
    "Burkina Faso":                 "burkina-faso",
    "South Sudan":                  "south-sudan",
    "Central African Republic":     "central-african-republic",
    "Bosnia-Herzegovina":           "bosnia-and-herzegovina",
    "North Korea":                  "north-korea",
    "South Korea":                  "south-korea",
    "Saudi Arabia":                 "saudi-arabia",
    "Sri Lanka":                    "sri-lanka",
    "Papua New Guinea":             "papua-new-guinea",
}
COUNTRY_TO_SLUG = {**auto_slugs, **SLUG_CORRECTIONS}

# Final list of (country_name, guardian_slug) pairs
COUNTRY_LIST = [(country, COUNTRY_TO_SLUG[country]) for country in acled_countries]

print(f"\nTotal countries to query: {len(COUNTRY_LIST)}")
print("\nSample mappings:")
for name, slug in COUNTRY_LIST[:5]:
    print(f"  {name} -> world/{slug}")


Total countries to query: 245

Sample mappings:
  Afghanistan -> world/afghanistan
  Akrotiri and Dhekelia -> world/akrotiri-and-dhekelia
  Albania -> world/albania
  Algeria -> world/algeria
  American Samoa -> world/american-samoa


Step 2: We define two collection functions: a standard version for most countries (fetch_articles), and a slower version with longer delays and timeout handling (fetch_articles_slow) for high-volume sections such as USA, UK, and Australia, where the standard function risks stalling during collection.

In [ ]:
BASE_URL    = "https://content.guardianapis.com/search"
current_key = 0  

def fetch_articles(country_name, slug):
    """
    Fetches all Guardian articles for a given country between FROM_DATE
    and TO_DATE. Paginates automatically and rotates API keys on rate
    limiting.

    For USA, UK, and Australia the Guardian uses dedicated top-level
    sections rather than world/ tags. These are detected by the "section:"
    prefix in the slug and queried using the section parameter instead of tag.

    Parameters:
        country_name : str — original ACLED country name, stored in output
        slug         : str — either "section:sectionid" or a world/ tag slug

    Returns a list of article dicts.
    """
    global current_key
    articles = []
    page = 1

    if slug.startswith("section:"):
        query_param = {"section": slug.replace("section:", "")}
    else:
        query_param = {"tag": f"world/{slug}"}

    while True:
        params = {
            **query_param,
            "from-date":   FROM_DATE,
            "to-date":     TO_DATE,
            "show-fields": "wordcount,sectionName,byline",
            "page-size":   200,
            "page":        page,
            "api-key":     API_KEYS[current_key]
        }

        r = requests.get(BASE_URL, params=params)
        time.sleep(0.1)

        if r.status_code == 429:
            current_key = (current_key + 1) % len(API_KEYS)
            print(f"  Rate limited — switching to key {current_key}")
            continue

        if r.status_code != 200:
            print(f"  Error {r.status_code} for {slug} page {page} — skipping")
            break

        data    = r.json()["response"]
        results = data.get("results", [])

        if not results:
            break

        for a in results:
            articles.append({
                "country":   country_name,
                "date":      a["webPublicationDate"],
                "section":   a["sectionName"],
                "wordcount": a.get("fields", {}).get("wordcount", None),
                "title":     a["webTitle"],
                "pillar":    a.get("pillarName", None),
                "byline": a.get("fields", {}).get("byline", None),
            })

        if page >= data["pages"]:
            break

        page += 1

    return articles


In [ ]:
def fetch_articles_slow(country_name, slug):
    """Same as fetch_articles but with longer sleep for large sections."""
    global current_key
    articles = []
    page = 1

    if slug.startswith("section:"):
        query_param = {"section": slug.replace("section:", "")}
    else:
        query_param = {"tag": f"world/{slug}"}

    while True:
        params = {
            **query_param,
            "from-date":   FROM_DATE,
            "to-date":     TO_DATE,
            "show-fields": "wordcount,sectionName,byline",
            "page-size":   200,
            "page":        page,
            "api-key":     API_KEYS[current_key]
        }

        try:
            r = requests.get(BASE_URL, params=params, timeout=30)
            time.sleep(0.5)  # slower — reduces stalling risk
        except requests.exceptions.Timeout:
            print(f"  Timeout on page {page} — retrying...")
            time.sleep(5)
            continue

        if r.status_code == 429:
            current_key = (current_key + 1) % len(API_KEYS)
            print(f"  Rate limited — switching to key {current_key}")
            time.sleep(2)
            continue

        if r.status_code != 200:
            print(f"  Error {r.status_code} on page {page} — stopping")
            break

        data    = r.json()["response"]
        results = data.get("results", [])
        if not results:
            break

        for a in results:
            articles.append({
                "country":   country_name,
                "date":      a["webPublicationDate"],
                "section":   a["sectionName"],
                "wordcount": a.get("fields", {}).get("wordcount", None),
                "title":     a["webTitle"],
                "pillar":    a.get("pillarName", None),
                "byline": a.get("fields", {}).get("byline", None),
            })

        if page % 50 == 0:
            print(f"  ...page {page}, {len(articles)} articles so far")

        if page >= data["pages"]:
            break

        page += 1

    return articles

Step 3: We now run the main collection loop for the listed ACLED countries making sure that if the raw data file already exists, it loads from disk rather than re-querying the API. This creates the guardian_raw.csv file.

In [ ]:
RAW_PATH = "../data/guardian_raw.csv"

if os.path.exists(RAW_PATH) and os.path.getsize(RAW_PATH) > 0:
    print("Guardian raw data already collected — loading from file.")
    guardian_raw = pd.read_csv(RAW_PATH)
else:
    all_articles = []
    for country_name, slug in COUNTRY_LIST:
        print(f"Fetching: {country_name}...")
        articles = fetch_articles(country_name, slug)
        all_articles.extend(articles)
        print(f"  -> {len(articles)} articles")

    guardian_raw = pd.DataFrame(all_articles)
    guardian_raw.to_csv(RAW_PATH, index=False)
    print(f"\nSaved {len(guardian_raw)} total articles to {RAW_PATH}")

print(f"\nShape: {guardian_raw.shape}")
print(f"Countries with articles: {guardian_raw['country'].nunique()}")
print(f"Date range: {guardian_raw['date'].min()} to {guardian_raw['date'].max()}")
guardian_raw.head()

Guardian raw data already collected — loading from file.

Shape: (276579, 7)
Countries with articles: 197
Date range: 2015-01-01T01:13:56Z to 2024-12-31T22:36:11Z


,country,date,section,wordcount,title,pillar,byline
0,Afghanistan,2024-12-31T18:41:27Z,UK news,495,Special forces troops face prosecution over al...,News,Kiran Stacey Political correspondent
1,Afghanistan,2024-12-29T17:09:11Z,Media,471,David Page obituary,News,William Crawley
2,Afghanistan,2024-12-23T17:00:21Z,US news,2008,‘Never underestimate what neighbors can do for...,News,David Wallis
3,Afghanistan,2024-12-06T17:50:37Z,Global development,446,Taliban move to ban women training as nurses a...,News,Ruchi Kumar and Zahra Joya for Rukhshana Media
4,Afghanistan,2024-12-06T10:40:29Z,Sport,471,Top Afghan cricketers urge Taliban to reverse ...,Sport,Jack Snape


Step 4: Due to the volume of articles in their dedicated Guardian sections, the USA, UK, and Australia are collected separately using the slower function and added to the guardian_raw.csv.

In [ ]:
aus_articles = fetch_articles_slow("Australia", "section:australia-news")
print(f"Australia: {len(aus_articles)} articles")

  ...page 50, 10000 articles so far
  ...page 100, 20000 articles so far
  ...page 150, 30000 articles so far
  Error 400 on page 191 — stopping
Australia: 38000 articles


In [ ]:
existing = pd.read_csv("../data/guardian_raw.csv")
aus_df = pd.DataFrame(aus_articles)
guardian_raw = pd.concat([existing, aus_df], ignore_index=True)
guardian_raw.to_csv("../data/guardian_raw.csv", index=False)
print(f"Total after Australia: {len(guardian_raw)} articles")

Total after Australia: 201282 articles


In [ ]:
uk_articles = fetch_articles_slow("United Kingdom", "section:uk-news")
print(f"UK: {len(uk_articles)} articles")

  ...page 50, 10000 articles so far
  ...page 100, 20000 articles so far
  ...page 150, 30000 articles so far
UK: 37297 articles


In [ ]:
existing = pd.read_csv("../data/guardian_raw.csv")
uk_df = pd.DataFrame(uk_articles)
guardian_raw = pd.concat([existing, uk_df], ignore_index=True)
guardian_raw.to_csv("../data/guardian_raw.csv", index=False)
print(f"Total after UK: {len(guardian_raw)} articles")

Total after UK: 238579 articles


In [ ]:
us_articles = fetch_articles_slow("United States", "section:us-news")
print(f"US: {len(us_articles)} articles")

  ...page 50, 10000 articles so far
  ...page 100, 20000 articles so far
  ...page 150, 30000 articles so far
  Error 400 on page 191 — stopping
US: 38000 articles


In [ ]:
existing = pd.read_csv("../data/guardian_raw.csv")
us_df = pd.DataFrame(us_articles)
guardian_raw = pd.concat([existing, us_df], ignore_index=True)
guardian_raw.to_csv("../data/guardian_raw.csv", index=False)
print(f"Total after USA: {len(guardian_raw)} articles")

Total after USA: 276579 articles


### 2.2 World Bank Income Classification

We load in the Ourworldindata world bank income dataset and use each country's classification to assign it a binary label: Global North (high income) or Global South (all others). We thus save the clasification to the worldbank_income.csv file.

In [ ]:
OWID_URL = (
    "https://ourworldindata.org/grapher/world-bank-income-groups.csv"
    "?v=1&csvType=full&useColumnShortNames=false"
)

wb_raw = pd.read_csv(
    OWID_URL,
    storage_options={"User-Agent": "Our World In Data data fetch/1.0"}
)

print("=== World Bank Income Classification ===")
print(f"Shape: {wb_raw.shape}")
print(f"Columns: {list(wb_raw.columns)}")
print(f"Years available: {wb_raw['Year'].min()} – {wb_raw['Year'].max()}")

wb_latest = (
    wb_raw
    .sort_values("Year", ascending=False)
    .drop_duplicates(subset="Entity")
    .rename(columns={"Entity": "country", "Year": "wb_year"})
)

wb_latest.to_csv("../data/worldbank_income.csv", index=False)
print(f"\nSaved to data/worldbank_income.csv")
print(f"Countries classified: {wb_latest['country'].nunique()}")
wb_latest.head()


=== World Bank Income Classification ===
Shape: (7953, 4)
Columns: ['Entity', 'Code', 'Year', "World Bank's income classification"]
Years available: 1987 – 2024

Saved to data/worldbank_income.csv
Countries classified: 226


,country,Code,wb_year,World Bank's income classification
4148,Liberia,LBR,2024,Low-income countries
7914,Zambia,ZMB,2024,Lower-middle-income countries
37,Afghanistan,AFG,2024,Low-income countries
4110,Lesotho,LSO,2024,Lower-middle-income countries
4186,Libya,LBY,2024,Upper-middle-income countries


We also applied manual overrides for geopolitically misclassified countries (e.g. Gulf states reclassified as Global South; Ukraine reclassified as Western) where their income classification doesn't reflect their geopolitical identity to the eyes of the world.

In [ ]:
INCOME_COL = [c for c in wb_latest.columns if "income" in c.lower()][0]
print(f"Using income column: '{INCOME_COL}'")

wb_latest["region_group"] = wb_latest[INCOME_COL].apply(
    lambda x: "Global North" if str(x).strip() == "High-income countries" else "Global South"
)
OVERRIDES = {
    "Ukraine":      "Global North",
    "Saudi Arabia": "Global South",
    "Qatar":        "Global South",
    "Kuwait":       "Global South",
    "Bahrain":      "Global South",
    "Oman":         "Global South",
    "United Arab Emirates": "Global South",
}

for country, label in OVERRIDES.items():
    mask = wb_latest["country"] == country
    if mask.any():
        wb_latest.loc[mask, "region_group"] = label
        print(f"  Override: {country} -> {label}")

# --- Build the two lists ---

global_north_countries = sorted(
    wb_latest[wb_latest["region_group"] == "Global North"]["country"].tolist()
)

global_south_countries = sorted(
    wb_latest[wb_latest["region_group"] == "Global South"]["country"].tolist()
)

print(f"Global North countries in ACLED:      {len(global_north_countries)}")
print(f"Global South countries in ACLED: {len(global_south_countries)}")
print(f"Global North sample:      {global_north_countries[:5]}")
print(f"Global South sample: {global_south_countries[:5]}")

wb_latest.to_csv("../data/worldbank_income.csv", index=False)
print("\nUpdated worldbank_income.csv with region_group column")

Using income column: 'World Bank's income classification'
  Override: Ukraine -> Global North
  Override: Saudi Arabia -> Global South
  Override: Qatar -> Global South
  Override: Kuwait -> Global South
  Override: Bahrain -> Global South
  Override: Oman -> Global South
  Override: United Arab Emirates -> Global South
Global North countries in ACLED:      85
Global South countries in ACLED: 141
Global North sample:      ['American Samoa', 'Andorra', 'Antigua and Barbuda', 'Aruba', 'Australia']
Global South sample: ['Afghanistan', 'Albania', 'Algeria', 'Angola', 'Argentina']

Updated worldbank_income.csv with region_group column


### 2.3 ACLED Fatalities

ACLED records all reported conflict fatalities by country and year. We use it as a simplified but internationally comparable severity proxy. We acknowledge that fatalities alone do not capture displacement, conflict duration, or full humanitarian scale. This limitation is discussed in the conclusion.

We import the downloadble csv file and transform into a clean pandas file, which is in turn saved into the cleaned_fatalities.csv file.

In [ ]:
cwd = os.getcwd()
data_path = Path.cwd().parent / "data"

fatalities = pd.read_csv(f"{data_path}//number_of_reported_fatalities_by_country-year_as-of-03Apr2026.csv")
fatalities = fatalities.reset_index()
fatalities.columns = fatalities.iloc[0].values.tolist()
fatalities = fatalities.drop(0).dropna(axis=1)
fatalities.to_csv(f"{data_path}//cleaned_fatalities.csv")
fatalities

,COUNTRY,YEAR,FATALITIES
1,Afghanistan,2017,36360
2,Afghanistan,2018,42991
3,Afghanistan,2019,41419
4,Afghanistan,2020,31037
5,Afghanistan,2021,42425
...,...,...,...
2939,eSwatini,2022,33
2940,eSwatini,2023,32
2941,eSwatini,2024,10
2942,eSwatini,2025,2


33 countries remain unmatched after corrections. On inspection, these are all small territories and overseas dependencies which have negligible conflict activity in the ACLED data and no meaningful Guardian coverage. They are excluded from the analysis without material impact on our findings.

# 3. Preparation of the Data for Analysis

We now have succesfully loaded our 3 datasets. However, to be able to move onto the EDA part of this project we must first complete some cleaning steps known as IDA. We will split this task in three sections for each dataset. Initial data analysis aims to develop a sound understanding of the data, and ensure the data is reliable, sufficient, and suitable for the intended analysis bef ore doing the actual analysis.

# IDA for guardian_raw file

We first check the basic structure of the dataset.

In [ ]:
print(guardian_raw.shape)
print(guardian_raw.dtypes)
print(guardian_raw.head())

(276579, 7)
country      object
date         object
section      object
wordcount     int64
title        object
pillar       object
byline       object
dtype: object
       country                  date             section  wordcount  \
0  Afghanistan  2024-12-31T18:41:27Z             UK news        495   
1  Afghanistan  2024-12-29T17:09:11Z               Media        471   
2  Afghanistan  2024-12-23T17:00:21Z             US news       2008   
3  Afghanistan  2024-12-06T17:50:37Z  Global development        446   
4  Afghanistan  2024-12-06T10:40:29Z               Sport        471   

                                               title pillar  \
0  Special forces troops face prosecution over al...   News   
1                                David Page obituary   News   
2  ‘Never underestimate what neighbors can do for...   News   
3  Taliban move to ban women training as nurses a...   News   
4  Top Afghan cricketers urge Taliban to reverse ...  Sport   

                            

We see that is a large dataset containing 276579 rows and 7 columns (country, date, section, wordcount, title pillar, byline). We notice correctly that every section has the "object" type except for wordcount which is an integer.

After seing the overall format of the dataset we must check for any missing values and for any duplicates among the data.

In [ ]:
print(guardian_raw.isnull().sum())
print(f"Wordcount missing: {guardian_raw['wordcount'].isnull().sum()} ({guardian_raw['wordcount'].isnull().mean()*100:.1f}%)")
print(f"Duplicate rows: {guardian_raw.duplicated().sum()}")
print(f"Duplicate titles: {guardian_raw.duplicated(subset='title').sum()}")
print("\nSample duplicate titles:")
print(guardian_raw[guardian_raw.duplicated(subset='title', keep=False)]
      ['title'].value_counts().head(10))

country         0
date            0
section         0
wordcount       0
title           0
pillar       1101
byline       2189
dtype: int64
Wordcount missing: 0 (0.0%)
Duplicate rows: 0
Duplicate titles: 50500

Sample duplicate titles:
title
What happened in the Russia-Ukraine war this week? Catch up with the must-read news and analysis    178
Coronavirus in the Pacific: weekly briefing                                                          54
European elections 2019: Tories unlikely to deliver on Brexit, says Farage – as it happened          27
There are reasons to be cheerful ... LGBTI rights gains in unlikely countries | Peter Tatchell       22
As Covid cases rise again, how are countries in Europe reacting?                                     20
Covid in Europe: how countries are tackling second wave                                              20
Saudi Arabia bans foreign pilgrims as Japan plans to close schools                                   19
Eurozone crisis: which countrie

We see thay 1101 Pillar values and 2189 byline values are missing. However, both of these absence are negligeable compared to the rest of the dataset as there is only 0.4% missing rate in the Pillar's section and 0.8% in the Byline's section case. 

Furthermore, we also notice that there are 50500 duplicate titles. Nevertheless, as we see in the sample duplicate titles section, thses aren't real duplicates as The Guardian uses recurring generic titles such as weekly briefings about Corona virus or the Ukraine war.

So the overall dataset has no real impactful missing values nor duplicate cases. We need to now make sure that the actual values of these articles make sense. 

We begin by examining the distribution of Guardian sections to check whether the articles collected correspond to the sections we would expect given our collection methodology.

In [ ]:
print(guardian_raw['section'].value_counts().head(20))

section
World news            84890
US news               43716
Australia news        42085
UK news               41032
Opinion               13245
Global development     8103
Business               5973
Environment            5564
Politics               4502
Film                   3848
Books                  2325
Art and design         1915
Media                  1656
Music                  1564
Technology             1482
Sport                  1481
News                   1479
Cities                 1442
Football               1392
Science                1067
Name: count, dtype: int64


World news dominates with 84,890 articles, as expected given we searched primarily via world/country tags. The remaining leading sections (US news, Australia news, UK news, and Opinion) reflect the three countries queried by section rather than tag, and the Guardian's substantial opinion output. This distribution is consistent with our collection methodology and raises no data quality concerns.

We now inspect the date formating to ensure coherent structure.

In [ ]:
print(guardian_raw['date'].sample(5))

24321     2019-05-12T15:46:14Z
131190    2024-11-27T15:43:10Z
78553     2017-12-22T14:56:46Z
235878    2015-09-16T06:00:20Z
211963    2022-05-28T15:34:55Z
Name: date, dtype: object


We see that the dates are stored in a consistent ISO format across all rows (eg. "year-month-dayThour:minute:secondZ") proving no anomalies are present. Moreover, these dates will have to be converted to datetime format in the cleaning process to enable the later time series analysis.

We now check the countrues with the lowest article count to make sure there are no anomalies.

In [ ]:
print(guardian_raw['country'].value_counts().tail(20))

country
Dominica                            26
Seychelles                          24
eSwatini                            23
Reunion                             23
Equatorial Guinea                   22
Togo                                22
Djibouti                            21
Saint Vincent and the Grenadines    20
Saint Kitts and Nevis               19
Turks and Caicos Islands            17
Suriname                            17
Cape Verde                          13
Guinea-Bissau                       13
Mayotte                             11
Montserrat                          10
San Marino                           9
Liechtenstein                        7
Andorra                              5
Comoros                              5
Sao Tome and Principe                5
Name: count, dtype: int64


The lowest-coverage countries are all small island states or micronations with minimal conflict activity. Andorra, San Marino, and Sao Tome and Principe each have fewer than 10 articles. These low counts reflect genuine lack of newsworthy conflict events rather than collection failures, and hence are consistent with their ACLED fatality figures.

Finally, we check the distribution of articles across years to ensure no anormal spikes in coverage.

In [ ]:
guardian_raw['year'] = pd.to_datetime(guardian_raw['date']).dt.year
print(guardian_raw['year'].value_counts().sort_index())

year
2015    24888
2016    21280
2017    24875
2018    25527
2019    24802
2020    26265
2021    26355
2022    35129
2023    33595
2024    33863
Name: count, dtype: int64


There is a clear jump in articles written in 2022 (35128 articles) from the previous years where they averaged around 24,000–26,000 articles per year. This is a persisting trend which follows into 2023 and 2024 and is almost certainly motivated by the Russian invasion of Ukraine in 2022 and the Israel-Gaza escalation in 2023. Thus, no anomalie has been detected.

# IDA for ACLED file (cleaned_fatalities)

We now move onto the ACLED cleaned_fatalities file.

Firstly, we check the overall structure of the dataset to confirm it loaded in correctly.

In [ ]:
print(acled.shape)
print(acled.dtypes)
print(acled.head())

(1634, 4)
COUNTRY         object
YEAR             int64
FATALITIES       int64
region_group    object
dtype: object
       COUNTRY  YEAR  FATALITIES  region_group
0  Afghanistan  2017       36360  Global South
1  Afghanistan  2018       42991  Global South
2  Afghanistan  2019       41419  Global South
3  Afghanistan  2020       31037  Global South
4  Afghanistan  2021       42425  Global South


The dataset contains 1,634 country-year observations across 4 columns. All types are as expected as country and region group are strings whereas year and fatalities are logically integers. The first rows confirm Afghanistan is correctly classified as Global South with plausible fatality figures across the study period.

Once again, after confirming the formatting of the file we verify wether there any missing or duplicate values. 

In [ ]:
print(acled.isnull().sum())
print(f"Duplicate country-year pairs: {acled.duplicated(subset=['COUNTRY','YEAR']).sum()}")

COUNTRY         0
YEAR            0
FATALITIES      0
region_group    0
dtype: int64
Duplicate country-year pairs: 0


In this case there are no identified missing values nor duplicate values.

We now examine the number of countries with recorded conflict activity per year to assess ACLED coverage consistency across the study window.

In [ ]:
print(acled['YEAR'].value_counts().sort_index())

YEAR
2015     61
2016     74
2017     77
2018    161
2019    161
2020    212
2021    245
2022    215
2023    213
2024    215
Name: count, dtype: int64


Coverage expands steadily from 61 countries in 2015 to over 200 from 2020 onwards, reflecting ACLED's improving global data collection rather than a genuine increase in conflict.

However, it is worth noting that this is a limitation as the Global South may thus have been under-represented in the earlier years. This is why the 2015-2024 window was chosen as it has a reasonable distribution of article counts across countries.

We now examine the numerical data surrounding the casualties to undestand our severity metric.

In [ ]:
print(acled['FATALITIES'].describe())
print(f"Zero fatality rows: {(acled['FATALITIES'] == 0).sum()}")

count     1634.000000
mean       991.755202
std       4549.874372
min          0.000000
25%          0.000000
50%          4.000000
75%        150.000000
max      73573.000000
Name: FATALITIES, dtype: float64
Zero fatality rows: 566


The fatality figures span a wide range, from 0 to 73573. The median is just 4 while the mean is 992, indicating a small number of extremely severe conflicts which drive up the overall average. 566 country-year rows record zero fatalities, meaning conflict events occurred but no deaths were reported. We also  beserve a maximum of 73573 casualties which is probably attributed to either Afghanistan or Syria at the height of their respective conflicts.

# IDA for World Bank income data 

Finally we will look over the World Bank income data.

We must first of all check the overall formatting of the data to ensure it has been correctly downloaded.

In [ ]:
wb_latest = wb_latest.reset_index(drop=True)
print(wb_latest.shape)
print(wb_latest[['country', INCOME_COL]].head(20))

(226, 5)
          country World Bank's income classification
0         Liberia               Low-income countries
1          Zambia      Lower-middle-income countries
2     Afghanistan               Low-income countries
3         Lesotho      Lower-middle-income countries
4           Libya      Upper-middle-income countries
5         Lebanon      Lower-middle-income countries
6   Liechtenstein              High-income countries
7          Latvia              High-income countries
8            Laos      Lower-middle-income countries
9       Lithuania              High-income countries
10     Kyrgyzstan      Lower-middle-income countries
11     Luxembourg              High-income countries
12         Kuwait              High-income countries
13          Macao              High-income countries
14         Kosovo      Upper-middle-income countries
15       Kiribati      Lower-middle-income countries
16     Madagascar               Low-income countries
17          Kenya      Lower-middle-i

The dataset contains 226 countries across 4 columns. The income classification column is present and contains the expected four categories: low income, lower-middle income, upper-middle income, and high income. We can thus be sure that the binary Global South/North classification will work.

We then inspect for missing values in the income classification and wether there are countries missing in the ACLED file from the World Bank Classification,.

In [ ]:
print(wb_latest[INCOME_COL].isnull().sum())
acled_names = set(acled['COUNTRY'].unique())
wb_names = set(wb_latest['country'].unique())
print("In ACLED but not World Bank:")
print(sorted(acled_names - wb_names))

0
In ACLED but not World Bank:
['Akrotiri and Dhekelia', 'Anguilla', 'Antarctica', 'Bailiwick of Guernsey', 'Bailiwick of Jersey', 'British Indian Ocean Territory', 'Caribbean Netherlands', 'Christmas Island', 'Cocos (Keeling) Islands', 'Cook Islands', 'Czech Republic', 'Falkland Islands', 'French Southern and Antarctic Lands', 'Guadeloupe', 'Heard Island and McDonald Islands', 'Ivory Coast', 'Martinique', 'Micronesia', 'Montserrat', 'Niue', 'Norfolk Island', 'Pitcairn', 'Republic of Congo', 'Reunion', 'Saint Helena, Ascension and Tristan da Cunha', 'Saint Pierre and Miquelon', 'Saint-Barthelemy', 'Saint-Martin', 'Sint Maarten', 'South Georgia and the South Sandwich Islands', 'Tokelau', 'United States Minor Outlying Islands', 'Vatican City', 'Virgin Islands, U.S.', 'Wallis and Futuna', 'eSwatini']


We see that there are no missing values. However, several countries present in ACLED do not appear in the World Bank classification, notably Ivory Coast, Czech Republic, and Republic of Congo. This is not because these countries are absent from the World Bank data, but because the two sources use different naming conventions for them. This issue will be addressed in the cleaning process.

Ultimately, to get a sense of the composition of our income classification we check its distribution

In [ ]:
print(wb_latest[INCOME_COL].value_counts())

World Bank's income classification
High-income countries            90
Upper-middle-income countries    59
Lower-middle-income countries    51
Low-income countries             26
Name: count, dtype: int64


High-income countries account for 90 entries, while the remaining 136 fall across upper-middle, lower-middle, and low income groups. Under our classification rule, these 136 form the Global South category. The majority of countries in the dataset are therefore classified as Global South, which aligns with the real-world distribution of nations.

# Cleaning Process

To assign a Global South or Western label to each country in the ACLED dataset, we merge it with the World Bank income classification.

To do so we have to make some manual country name harmonisation changes as they don't appear the same on both files eg. World Bank lists "Czechia" while ACLED uses "Czech Republic". We correct these mismatches before applying the classification, then save the result to cleaned_fatalities.csv.

In [ ]:
data_path = Path.cwd().parent / "data"

work_bank_df = pd.read_csv(
    "https://ourworldindata.org/grapher/world-bank-income-groups.csv"
    "?v=1&csvType=full&useColumnShortNames=false",
    storage_options={"User-Agent": "Our World In Data data fetch/1.0"}
)

print("World Bank shape:", work_bank_df.shape)
print(work_bank_df.head())

WB_TO_ACLED = {
    "Congo":        "Republic of Congo",       # was "Congo, Rep."
    "Czechia":      "Czech Republic",
    "Eswatini":     "eSwatini",
    "Congo, Dem. Rep.":  "Democratic Republic of Congo",
    "Côte d'Ivoire":     "Ivory Coast",
    "Korea, Rep.":       "South Korea",
    "Korea, Dem. People's Rep.": "North Korea",
    "Lao PDR":           "Laos",
    "Syrian Arab Republic": "Syria",
    "Yemen, Rep.":       "Yemen",
    "Venezuela, RB":     "Venezuela",
    "Iran, Islamic Rep.": "Iran",
    "Egypt, Arab Rep.":  "Egypt",
    "Kyrgyz Republic":   "Kyrgyzstan",
    "Turkiye":           "Turkey",
}

work_bank_df["Entity"] = work_bank_df["Entity"].replace(WB_TO_ACLED)

INCOME_COL = "World Bank's income classification"

wb_latest = (
    work_bank_df
    .sort_values("Year", ascending=False)
    .drop_duplicates(subset="Entity")
)

global_south = wb_latest.loc[
    wb_latest[INCOME_COL] != 'High-income countries',
    'Entity'
].unique().tolist()

western = wb_latest.loc[
    wb_latest[INCOME_COL] == 'High-income countries',
    'Entity'
].unique().tolist()

OVERRIDES_WESTERN = ["Ukraine"]
OVERRIDES_GLOBAL_SOUTH = [
    "Saudi Arabia", "Qatar", "Kuwait",
    "Bahrain", "Oman", "United Arab Emirates"
]

for c in OVERRIDES_WESTERN:
    if c not in western:
        western.append(c)
    if c in global_south:
        global_south.remove(c)

for c in OVERRIDES_GLOBAL_SOUTH:
    if c not in global_south:
        global_south.append(c)
    if c in western:
        western.remove(c)

print(f"\nGlobal South countries: {len(global_south)}")
print(f"Western countries:      {len(western)}")

fatalities = pd.read_csv(
    f"{data_path}/number_of_reported_fatalities_by_country-year_as-of-03Apr2026.csv",
    skiprows=1
)

fatalities = fatalities.loc[:, ~fatalities.columns.str.contains('^Unnamed')]

print("\nFatalities shape:", fatalities.shape)
print(fatalities.head())

# Filter to study window 2015-2024 only
# ACLED has data from 1997 but coverage is sparse before 2015.
# We also exclude 2025 and 2026 which are outside our Guardian date range.
fatalities = fatalities[
    (fatalities['YEAR'] >= 2015) &
    (fatalities['YEAR'] <= 2024)
]

print(f"\nAfter date filter: {fatalities.shape}")
print(fatalities['YEAR'].value_counts().sort_index())

# =============================================================================
# ADD REGION CLASSIFICATION TO FATALITIES
# =============================================================================

fatalities["region_group"] = fatalities["COUNTRY"].apply(
    lambda x: "Western" if x in western else "Global South"
)

print("\nRegion group distribution:")
print(fatalities["region_group"].value_counts())

# Sanity check — countries not matched to either list
unmatched = fatalities[
    ~fatalities["COUNTRY"].isin(western + global_south)
]["COUNTRY"].unique()
print(f"\nUnmatched countries: {len(unmatched)}")
if len(unmatched) > 0:
    print(sorted(unmatched))

# =============================================================================
# SAVE CLEAN VERSION
# =============================================================================

fatalities.to_csv(f"{data_path}/cleaned_fatalities.csv", index=False)
print(f"\nSaved cleaned_fatalities.csv — shape: {fatalities.shape}")

World Bank shape: (7953, 4)
        Entity Code  Year World Bank's income classification
0  Afghanistan  AFG  1987               Low-income countries
1  Afghanistan  AFG  1988               Low-income countries
2  Afghanistan  AFG  1989               Low-income countries
3  Afghanistan  AFG  1990               Low-income countries
4  Afghanistan  AFG  1991               Low-income countries

Global South countries: 141
Western countries:      85

Fatalities shape: (2943, 3)
       COUNTRY  YEAR  FATALITIES
0  Afghanistan  2017       36360
1  Afghanistan  2018       42991
2  Afghanistan  2019       41419
3  Afghanistan  2020       31037
4  Afghanistan  2021       42425

After date filter: (1634, 3)
YEAR
2015     61
2016     74
2017     77
2018    161
2019    161
2020    212
2021    245
2022    215
2023    213
2024    215
Name: count, dtype: int64

Region group distribution:
region_group
Global South    1221
Western          413
Name: count, dtype: int64

Unmatched countries: 33
['Akroti

Furthermore, as identified in the IDA, the date column is stored as a string and requires conversion before analysis. We also extract year and month from the date column to enable time series aggregation later.

In [ ]:
guardian_raw['date'] = pd.to_datetime(guardian_raw['date'], utc=True)

guardian_raw['year'] = guardian_raw['date'].dt.year
guardian_raw['month'] = guardian_raw['date'].dt.month

print(guardian_raw.dtypes)
print(guardian_raw.isnull().sum())
print(guardian_raw.shape)

country                   object
date         datetime64[ns, UTC]
section                   object
wordcount                  int64
title                     object
pillar                    object
byline                    object
year                       int32
month                      int32
dtype: object
country         0
date            0
section         0
wordcount       0
title           0
pillar       1101
byline       2189
year            0
month           0
dtype: int64
(276579, 9)


# Making the clean dataset

Finally, to facilitate the later analysis we create two datasets. master.csv contains one row per country, aggregating total article count, average and median word count, total fatalities, section placement breakdown, and a normalised coverage intensity metric (articles per 1,000 fatalities). master_monthly.csv contains one row per country per month, enabling time series and responsiveness analysis. Both are saved to the /data folder.

In [ ]:
guardian_raw['wordcount'] = pd.to_numeric(guardian_raw['wordcount'], errors='coerce')

guardian_agg = guardian_raw.groupby('country').agg(
    total_articles = ('title',     'count'),
    avg_wordcount  = ('wordcount', 'mean'),
    med_wordcount  = ('wordcount', 'median'),
).reset_index()

acled_agg = acled.groupby('COUNTRY').agg(
    total_fatalities = ('FATALITIES', 'sum'),
    region_group     = ('region_group', 'first')
).reset_index().rename(columns={'COUNTRY': 'country'})

master = guardian_agg.merge(acled_agg, on='country', how='inner')

section_breakdown = (
    guardian_raw
    .groupby(['country', 'section'])
    .size()
    .reset_index(name='count')
)
total_per_country = guardian_raw.groupby('country').size().reset_index(name='total')
section_breakdown = section_breakdown.merge(total_per_country, on='country')
section_breakdown['pct'] = section_breakdown['count'] / section_breakdown['total'] * 100

section_pivot = section_breakdown.pivot_table(
    index='country', columns='section', values='pct', fill_value=0
).reset_index()

master = master.merge(section_pivot, on='country', how='left')

master['articles_per_1000_fatalities'] = (
    master['total_articles'] / master['total_fatalities'] * 1000
).replace([float('inf'), -float('inf')], None)

print("=== Master Dataset ===")
print(f"Shape: {master.shape}")
print(f"Countries: {master['country'].nunique()}")
print(f"\nRegion split:\n{master['region_group'].value_counts()}")
print(f"\nSample:\n{master[['country', 'region_group', 'total_articles', 'total_fatalities', 'avg_wordcount']].head(10)}")

master.to_csv("../data/master.csv", index=False)
print("\nSaved master.csv")

guardian_raw['date'] = pd.to_datetime(guardian_raw['date'], utc=True)
guardian_raw['year']  = guardian_raw['date'].dt.year
guardian_raw['month'] = guardian_raw['date'].dt.month

guardian_monthly = guardian_raw.groupby(
    ['country', 'year', 'month']
).agg(
    article_count = ('title',     'count'),
    avg_wordcount = ('wordcount', 'mean')
).reset_index()

acled_monthly = acled.copy()
acled_monthly['monthly_fatalities'] = acled_monthly['FATALITIES'] / 12
acled_monthly = acled_monthly.rename(columns={'COUNTRY': 'country', 'YEAR': 'year'})

monthly = guardian_monthly.merge(
    acled_monthly[['country', 'year', 'monthly_fatalities', 'region_group']],
    on=['country', 'year'],
    how='left'
)

print("=== Monthly Dataset ===")
print(f"Shape: {monthly.shape}")
print(monthly.head(10))

monthly.to_csv("../data/master_monthly.csv", index=False)
print("\nSaved master_monthly.csv")


=== Master Dataset ===
Shape: (197, 69)
Countries: 197

Region split:
region_group
Global South    129
Western          68
Name: count, dtype: int64

Sample:
               country  region_group  total_articles  total_fatalities  \
0          Afghanistan  Global South            2656            200899   
1              Albania  Global South             162                 4   
2              Algeria  Global South             130               649   
3              Andorra       Western               5                 0   
4               Angola  Global South             117               765   
5           Antarctica  Global South             434                 0   
6  Antigua and Barbuda       Western              46                 6   
7            Argentina  Global South             736               120   
8              Armenia  Global South             185               297   
9            Australia       Western           38000                 7   

   avg_wordcount  
0     98

# 4. Exploratory Data Analysis

Question 3: Does The Guardian respond to casualty spikes differently depending on region, and has any disparity narrowed or widened over time (2015–2024)?

To answer this question, we will first check how many countries are both in top 10 for casualties and the Guardian articles, then, we will find their pearson correlation coefficient to measures strength of linear relationship between these two variable in each region, and see how does this correlation changed over time from 2015 to 2024. The reason why we choose Pearson correlation is that we are testing linear relationsip between these variable and both of them are continous and quantitativ measures. In addition to that, as we also intend to compare the responsiveness in each region, Pearson correlation is a good idea as it also show us the strength of the relationship

Step 1: Preparing data for our analysis


In [ ]:
monthly = pd.read_csv('..\data\master_monthly.csv')
master  = pd.read_csv('..\data\master.csv')
wb      = pd.read_csv('..\data\worldbank_income_clean.csv')
 
wb_region = wb[['country', 'region_group']].drop_duplicates()
 
# Join monthly
monthly = monthly.drop(columns='region_group')
monthly = monthly.merge(wb_region, on='country', how='inner')
 
# Join master
master = master.drop(columns='region_group')
master = master.merge(wb_region, on='country', how='inner')

df = monthly[monthly['monthly_fatalities'].notna()].copy()

yearly_agg = df.groupby(['year', 'region_group']).agg(
    total_articles   = ('article_count',     'sum'),
    total_fatalities = ('monthly_fatalities', 'sum')
).reset_index()

Step 2: Prepare graph plotting function for analysis

In [ ]:
def axis_ref(col):
    return "x domain" if col == 1 else f"x{col} domain"
 
def yaxis_ref(col):
    return "y domain" if col == 1 else f"y{col} domain"
 
 
def build_scatter(master_data, region, color, title):
    """
    Scatter plot of casualties vs articles for one region.
    Prints Pearson r when called.
    """
    sub = master_data[
        (master_data['region_group'] == region) &
        (master_data['total_fatalities'].notna()) &
        (master_data['total_fatalities'] > 0)
    ].copy()
 
    r_val, p_val = pearsonr(sub['total_fatalities'], sub['total_articles'])
    print(f"\n{region} country-level Pearson: r = {r_val:.4f}, p = {p_val:.4e}, n = {len(sub)}")
 
    fig = go.Figure()
 
    fig.add_trace(go.Scatter(
        x=sub['total_fatalities'], y=sub['total_articles'],
        mode='markers+text',
        marker=dict(color=color, size=8, opacity=0.7),
        text=sub['country'],
        textposition='top center',
        textfont=dict(size=8),
        hovertemplate=(
            '<b>%{text}</b><br>'
            'Casualties: %{x:,.0f}<br>'
            'Articles: %{y:,}<extra></extra>'
        )
    ))
 
    # OLS trend line
    z = np.polyfit(sub['total_fatalities'], sub['total_articles'], 1)
    x_line = np.linspace(sub['total_fatalities'].min(), sub['total_fatalities'].max(), 100)
    fig.add_trace(go.Scatter(
        x=x_line, y=np.polyval(z, x_line),
        mode='lines',
        line=dict(color=color, width=2, dash='dash'),
        showlegend=False
    ))
 
    fig.add_annotation(
        text=f"r = {r_val:.3f}, p = {p_val:.2e}, n = {len(sub)}",
        xref="x domain", yref="y domain",
        x=0.95, y=0.95,
        xanchor='right', yanchor='top',
        showarrow=False,
        font=dict(size=14),
        bgcolor='rgba(255,255,255,0.8)',
        bordercolor=color, borderwidth=1
    )
 
    fig.update_layout(
        title=title,
        xaxis_title='Total Casualties',
        yaxis_title='Total Articles',
        template='plotly_white',
        width=1000, height=600,
        showlegend=False
    )
 
    return fig
 
 
def build_top10_bars(master_data, region, color_art, color_cas, title):
    """
    Two horizontal bar charts stacked vertically:
      Top: top 10 by article count
      Bottom: top 10 by casualty count
    """
    sub = master_data[
        (master_data['region_group'] == region) &
        (master_data['total_fatalities'].notna()) &
        (master_data['total_fatalities'] > 0)
    ]
 
    top_articles = sub.nlargest(10, 'total_articles').sort_values('total_articles')
    top_casualties = sub.nlargest(10, 'total_fatalities').sort_values('total_fatalities')
 
    fig = make_subplots(
        rows=2, cols=1,
        subplot_titles=(
            f'{region}: Top 10 by Article Count',
            f'{region}: Top 10 by Casualty Count'
        ),
        vertical_spacing=0.15
    )
 
    fig.add_trace(go.Bar(
        y=top_articles['country'],
        x=top_articles['total_articles'],
        orientation='h',
        marker_color=color_art,
        text=top_articles['total_articles'].apply(lambda x: f'{x:,}'),
        textposition='outside',
        hovertemplate='<b>%{y}</b><br>Articles: %{x:,}<extra></extra>'
    ), row=1, col=1)
 
    fig.add_trace(go.Bar(
        y=top_casualties['country'],
        x=top_casualties['total_fatalities'],
        orientation='h',
        marker_color=color_cas,
        text=top_casualties['total_fatalities'].apply(lambda x: f'{x:,.0f}'),
        textposition='outside',
        hovertemplate='<b>%{y}</b><br>Casualties: %{x:,.0f}<extra></extra>'
    ), row=2, col=1)
    
    art_max = top_articles['total_articles'].max()
    cas_max = top_casualties['total_fatalities'].max()
    
    fig.update_xaxes(title_text='Total Articles', range=[0, art_max * 1.25], row=1, col=1)
    fig.update_xaxes(title_text='Total Casualties', range=[0, cas_max * 1.25], row=2, col=1)
 
    fig.update_layout(
        title=title,
        template='plotly_white',
        width=900, height=700,
        showlegend=False
    )
 
    return fig
 
 
def build_pearson_by_year(monthly_data):
    """
    Line chart of Pearson r by year for both regions.
    Prints correlation stats when called.
    """
    df = monthly_data[monthly_data['monthly_fatalities'].notna()].copy()
 
    yearly = df.groupby(['country', 'year', 'region_group']).agg(
        total_articles   = ('article_count',     'sum'),
        total_fatalities = ('monthly_fatalities', 'sum')
    ).reset_index()
 
    results = []
    for region in ['Global South', 'Global North']:
        for year in sorted(yearly['year'].unique()):
            sub = yearly[(yearly['region_group'] == region) & (yearly['year'] == year)]
            if len(sub) > 2:
                r_val, p_val = pearsonr(sub['total_fatalities'], sub['total_articles'])
                countries = sorted(sub['country'].tolist())
                results.append({
                    'year': year,
                    'region_group': region,
                    'r': r_val,
                    'p': p_val,
                    'n': len(sub),
                    'significant': p_val < 0.05,
                    'countries': countries
                })
 
    results_df = pd.DataFrame(results)
 
    print(f"\n{'='*60}")
    print("PEARSON r BY YEAR")
    print(f"{'='*60}")
    for _, row in results_df.iterrows():
        sig = '***' if row['p'] < 0.001 else '**' if row['p'] < 0.01 else '*' if row['p'] < 0.05 else ''
        print(f"  {row['region_group']:15s} {row['year']}  r={row['r']:.3f}  p={row['p']:.4e}  n={row['n']:>3} {sig}")
 
    gs = results_df[results_df['region_group'] == 'Global South']
    gn = results_df[results_df['region_group'] == 'Global North']
 
    fig = go.Figure()
 
    for subset, name, color in [(gs, 'Global South', '#E07A5F'), (gn, 'Global North', '#3D405B')]:
        hover_texts = []
        for _, row in subset.iterrows():
            c_list = row['countries']
            if len(c_list) > 15:
                c_str = ', '.join(c_list[:15]) + f'... (+{len(c_list)-15} more)'
            else:
                c_str = ', '.join(c_list)
            hover_texts.append(
                f"<b>{name} {row['year']}</b><br>"
                f"r = {row['r']:.3f}, p = {row['p']:.2e}<br>"
                f"n = {row['n']} countries<br>"
                f"<br>{c_str}"
            )
 
        fig.add_trace(go.Scatter(
            x=subset['year'], y=subset['r'],
            name=name,
            mode='lines+markers+text',
            line=dict(color=color, width=2),
            marker=dict(
                size=10,
                symbol=subset['significant'].apply(
                    lambda s: 'circle' if s else 'circle-open'
                ),
                color=color
            ),
            text=subset['n'].apply(lambda n: f'n={n}'),
            textposition='top center',
            textfont=dict(size=9, color=color),
            hovertemplate='%{customdata}<extra></extra>',
            customdata=hover_texts
        ))
 
    fig.add_hline(y=0, line_dash='dot', line_color='grey', opacity=0.5)
 
    fig.update_layout(
        title='Pearson Correlation (Casualties vs Articles) by Year and Region',
        xaxis_title='Year',
        yaxis_title='Pearson r',
        margin=dict(b=80),
        yaxis=dict(range=[-0.3, 1.0]),
        xaxis=dict(dtick=1),
        template='plotly_white',
        width=1000, height=550,
        legend=dict(x=0.01, y=0.01, xanchor='left', yanchor='bottom',
                    bgcolor='rgba(255,255,255,0.8)'),
        annotations=[dict(
            text='● filled = significant (p < 0.05)   ○ open = not significant',
            xref='paper', yref='paper',
            x=0.5, y=-0.18, showarrow=False,
            font=dict(size=11, color='grey')
        )]
    )
 
    return fig

def build_yearly_scatter(yearly, title):
    """
    Side-by-side scatter: Global South | Global North
    One point per year, with Pearson r.
    Prints stats when called.
    """
    fig = make_subplots(
        rows=1, cols=2,
        subplot_titles=('Global South', 'Global North'),
        horizontal_spacing=0.12
    )

    print(f"\n{'='*60}")
    print(f"PEARSON – {title}")
    print(f"{'='*60}")

    for col, (region, color) in enumerate([
        ('Global South', '#E07A5F'),
        ('Global North', '#3D405B')
    ], start=1):
        sub = yearly[yearly['region_group'] == region]
        if len(sub) < 3:
            continue

        r_val, p_val = pearsonr(sub['total_fatalities'], sub['total_articles'])
        print(f"  {region}:  r = {r_val:.4f}, p = {p_val:.4e}, n = {len(sub)}")

        fig.add_trace(go.Scatter(
            x=sub['total_fatalities'], y=sub['total_articles'],
            mode='markers+text',
            marker=dict(color=color, size=10),
            text=sub['year'].astype(str),
            textposition='top center',
            textfont=dict(size=9),
            hovertemplate=(
                'Year: %{text}<br>'
                'Casualties: %{x:,.0f}<br>'
                'Articles: %{y:,}<extra></extra>'
            )
        ), row=1, col=col)

        z = np.polyfit(sub['total_fatalities'], sub['total_articles'], 1)
        x_line = np.linspace(sub['total_fatalities'].min(),
                             sub['total_fatalities'].max(), 100)
        fig.add_trace(go.Scatter(
            x=x_line, y=np.polyval(z, x_line),
            mode='lines',
            line=dict(color=color, width=2, dash='dash'),
            showlegend=False
        ), row=1, col=col)

        fig.add_annotation(
            text=f"r = {r_val:.3f}, p = {p_val:.2e}",
            xref=axis_ref(col), yref=yaxis_ref(col),
            x=0.95, y=0.95,
            xanchor='right', yanchor='top',
            showarrow=False,
            font=dict(size=13),
            bgcolor='rgba(255,255,255,0.8)',
            bordercolor=color, borderwidth=1
        )

    fig.update_xaxes(title_text='Yearly Casualties', row=1, col=1)
    fig.update_xaxes(title_text='Yearly Casualties', row=1, col=2)
    fig.update_yaxes(title_text='Yearly Articles', row=1, col=1)

    fig.update_layout(
        title=title,
        template='plotly_white',
        width=1000, height=500,
        showlegend=False
    )

    return fig

Step 3: We will now plot graph for our analysis

Graph 1

In [ ]:
fig_yearly_scatter = build_yearly_scatter(
    yearly_agg,
    'Yearly: The Guardian articles vs ACLED casualties by region '
)
fig_yearly_scatter.show()

The scatter graph show that these two variables have stronger correlation in Global North than in Global South, with the pearson correlation of 0.7856 compare to 0.3910. Also, consider the p-value of Global South which is 0.26 compared to 0.012 of Global North, we can conclude that it was statistically significant that casualties have positive correlation to number of articles in Global North than in Global South from 2015 to 2024



Graph 2

In [ ]:
fig_gs_bars = build_top10_bars(
    master, 'Global South', '#E07A5F', '#F2CC8F',
    'Global South: Top 10 Countries by Coverage vs Casualties (2015–2024)'
)
fig_gs_bars.show()

fig_gs_scatter = build_scatter(
    master, 'Global South', '#E07A5F',
    'Global South: Total Casualties vs Total Articles by Country (2015–2024)'
)
fig_gs_scatter.show()

As we already found out there was correlation between casualties and articles in Global South, we first find out whether top 10 countries with highest casualties were also top 10 countries with highest number of articles. The bar chart show that 5/10 countries appeared in both list

To find out whether the correlation still hold, we will also plot scatter graph between casualties and articles by country.

The above plot show there was statistically significant medium positive correlation of 0.330, indicates the Guardian actually did respond to casualties spike in this region

Graph 3

In [ ]:
fig_gn_bars = build_top10_bars(
    master, 'Global North', '#3D405B', '#81B29A',
    'Global North: Top 10 Countries by Coverage vs Casualties (2015–2024)'
)
fig_gn_bars.show()

fig_gn_scatter = build_scatter(
    master, 'Global North', '#3D405B',
    'Global North: Total Casualties vs Total Articles by Country (2015–2024)'
)
fig_gn_scatter.show()

When we do the same thing for the Global North, even though 6/10 countries appeared in both list, the number wasn't as coherent like in Global South analysis. 

Surpringsingly, this graph show different result compare to our first graph. This scatter graph show that the correlation between casualties and articles in Global North was statistically insignificant and very weak. Hence, this show that the correlation we found in first graph was just coincident and Global North related articles might be driven by other reason such as geopolitical or reason related to the demographic of reader rather than in respond to casualty spike

To find out how responsive of the Guardian toward casualties changed overtime, we plot line graph show how pearson correlation changed by year. Noted that at some year, some countries have no casualties data due to ACLED limitation, so the number of countries included in each region each year is different

Graph 4

In [ ]:
fig_pearson = build_pearson_by_year(monthly)
fig_pearson.show()
 



We can see that the Guardian is less responsive toward casualties spike in Global South over the year, with only exception of a spike in 2021. Meanwhile, in Global North, the Guardian responsiveness toward casualties spiked relatively unchanged, except in 2020 and in 2022, which was the start of the Russia - Ukraine war



However, as we saw that there was some outliers in the data such as China in Global South or US, UK and Australia in Global North with high number of articles and low number of casualties, as these country have significant geopolitical influence or where most of the Guardian readers based in. In addition to that, these outliers also violated the assumption of the Pearson correlation, which may affected to it's behaviour.



In [ ]:
EXCLUDE_GS = ['China']
EXCLUDE_GN = ['United Kingdom', 'United States', 'Australia']

monthly_no_outliers = monthly[~(
    ((monthly['region_group'] == 'Global South') & (monthly['country'].isin(EXCLUDE_GS))) |
    ((monthly['region_group'] == 'Global North') & (monthly['country'].isin(EXCLUDE_GN)))
)]

fig_pearson_no_outliers = build_pearson_by_year(monthly_no_outliers)
fig_pearson_no_outliers.update_layout(
    title='Pearson r by Year (Excl. China for GS; UK, US, Australia for GN)'
)
fig_pearson_no_outliers.show()

As we can see, when we exclude these outliers, the result actually became very different. Now, all the result for Global South are statistically significant and from 2022-2024, the pearson correlation for Global North region were statistically significant. In addition to that, the pearson correlation also increased compared to when we included those outlier, suggest, the Guardian were responsive to casualties when we exclude country where a lots of the Guardian reader lived or country with significant geopolitical influence like China

The graph did show a different result, as all of the pearson correlation between these variables in Global South were stronger and statistically significant. Meanwhile, in Global North, the correlation also became statistically significant from 2022 to 2024 and stronger. 

# 5. Conclusion